checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.checker;
+
+import alluxio.PropertyKey;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.SparkConf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import scala.Tuple2;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A spark job to test the integration of Spark with Alluxio. This class will be triggered
+ * in yarn-client and yarn-cluster mode through spark-submit.
+ *
+ * This job requires a running Spark cluster, but not requires a running Alluxio cluster.
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized in Spark
+ * driver and executors.
+ */
+public class SparkIntegrationChecker {
+
+  @Parameter(names = {"-p", "--partition"}, description = "The user defined partition number")
+  private static int sPartition;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkIntegrationChecker.class);
+  private static List<Tuple2<Integer, String>> sSparkJobResult;
+  private static boolean sAlluxioHAMode = false;
+  private static String sZookeeperAddress = "";
+
+  /**
+   * Implements check integration of Spark with Alluxio.
+   *
+   * @param sc current JavaSparkContext
+   * @return 0 succeed, 1 fail to recognize Alluxio classes,
```

Comment:
returning an int is not ideal here since you still need to explain what each integer value means in the javadoc. A better practice is to create an enmu class which is more informative

Reply:
That's a good idea! I will change it to an enmu class.